Implemented "move container to workspace"

minus7 opened the pull request:
- Fixes to the visibility of views
- `move container to workspace <name>`
- rewrote `workspace{,_output}_{prev,next}` and patched that into `workspace_by_name` so it works with the move command

minus7 commented:
You can totally ignore the first commit, it was all changed in the last one again.

SirCmpwn commented on include/container.h:
```diff
@@ -56,6 +56,11 @@ struct sway_container {
+enum view_visibility {
+	INVISIBLE = 1,
+	VISIBLE = 2
+};
```
Why is a boolean not sufficient

minus7 replied:
I kept it how it was, just changing it to use constants. Could add additional visibility layers there in future, is what i thought.

I'll change it to be a bool only.

SirCmpwn commented:
Please confirm that this works for the given test cases:

* Move container to workspace on same output
* Move container to same workspace it's already on
* Move container to workspace that is visible on a different output
* Move container to a workspace that is not visible on a different output

SirCmpwn commented:
Update the man pages

minus7 commented:
Addressed all issues except for one thing: Outputs WLC-0 with workspace 1 focused, WLC-1 with workspace 2 focused and workspace 3 hidden with one view on it. Moving a view from 1 to 3 works as expected, but when you then show 3, the moved over view has ⅓ of the output width instead of half of it. Must be a bug in arrange_windows which I didn't touch.

SirCmpwn commented on include/container.h:
```diff
@@ -56,6 +56,10 @@ struct sway_container {
 
+enum visibility_mask {
+	VISIBLE = 1
+};
```
Why enum instead of boolean
